src/components/Modal/__stories__/modal.stories.mdx

```diff
@@ -0,0 +1,308 @@
+import Modal from "../Modal";
+import { useModalHelper } from "./helpers";
+import { ArgsTable, Story, Canvas, Meta } from "@storybook/addon-docs";
+import { createStoryMetaSettings } from "../../../storybook/functions/create-component-story";
+import "./Modal.stories.scss";
+import { ModalContent, ModalFooter, ModalHeader, TextField } from "components";
+import { Settings } from "components/Icon/Icons";
+
+export const metaSettings = createStoryMetaSettings({
+  component: Modal,
+  enumPropNamesArray: [], // List enum props here
+  iconPropNamesArray: [], // List props that are typed as icons here
+  actionPropsArray: [] // List the component's actions here
+});
+
+<Meta title="Feedback/Modal" component={Modal} argTypes={metaSettings.argTypes} decorators={metaSettings.decorators} />
+
+<!--- Component documentation -->
+
+# Modal
+
+- [Overview](#overview)
+- [Props](#props)
+- [Usage](#usage)
+- [Variants](#variants)
+- [Do’s and don’ts](#dos-and-donts)
+- [Use cases and examples](#use-cases-and-examples)
+- [Related components](#related-components)
+- [Feedback](#feedback)
+
+## Overview
+
+An easy to use accessible Modal component
+
+<Canvas>
+  <Story name="Overview">
+    {() => {
+      const { openModalButton, modalButtons, show, closeModal, ref } = useModalHelper();
+      return (
+        <div>
+          {openModalButton}
+          <Modal
+            id={"story-book-modal"}
+            title={"Modal title"}
+            triggerElement={ref.current}
+            show={show}
+            onClose={closeModal}
+            closeButtonAriaLabel={"close"}
+          >
+            <ModalContent>Modal content goes here</ModalContent>
+            <ModalFooter>{modalButtons}</ModalFooter>
+          </Modal>
+        </div>
+      );
+    }}
+  </Story>
+</Canvas>
+
+## Props
+
+<ArgsTable of={Modal} />
+
+## Usage
+
+### Showing the Modal
+
+You show/hide the Modal by passing the <b>show</b> prop. This gives you complete control on when to show and hide the Modal.
+One thing to consider is when the user initiate closing the Modal by
+
+### Modal content
+
+### Design and customizations
+
+<UsageGuidelines guidelines={["Usage guideline 1", "Usage guideline 2", "Usage guideline 3"]} />
+
+<Tip title="Not what you were looking for?">Tip content</Tip>
+
+## Variants
+
+### Alert Modal (Modal)
+
+This Modal will NOT be closed when you click outside or click ESC key. Only way to close it is to use the close button.
+
+<UsageGuidelines guidelines={["set `isAlertDialog` to `true` in order to turn a Modal to modal."]} />
+
+<Canvas>
+  <Story name="Alert Modal">
```

Comment:
Story stop working for me when I open modal - close with Esc - and then try to open once again - Modal not opening.